### PR TITLE
Assignment 3

### PR DESCRIPTION
**Pull request by pnotequalnp**

### CODE REVIEW

**Comment by roonle** [comment]
Ill review this!

File: SecretSanta/src/SecretSanta.Business/GiftService.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public class GiftService : EntityService<Gift>, IEntityService<Gift>
+    {
+        public GiftService(ApplicationDbContext dbContext, IMapper mapper) :
+            base(dbContext, mapper) { }
+
+        public override async Task<Gift> FetchByIdAsync(int id) =>
+          await ApplicationDbContext
+            .Set<Gift>()
+            .Include(nameof(Gift.User))
```
Comment (roonle) [inline SecretSanta/src/SecretSanta.Business/GiftService.cs]: nice pre-populated Gift user here

**Comment by roonle** [comment]
Assignment
Important Note: Assignment due dates are one day later for Assignment 3

Connect Azure DevOps up for your fork.
Add a Status Badge (see Add a status badge to your repository) to the README.md file in master that shows the build status on each branch. ✔
Be sure that clicking on the badge successfully navigates one to the Azure DevOps pipeline. ✔
Rebase to the Readme.md update on master to the Assignment3 branch so it too shows the badge (be sure it points to the Assignment3 branch status) ✔
Appropriately encapsulate Id and all fingerprint data. ✔
Add non-default constructors to Gift and address all warnings (see BlogEngine's Post.cs file). ✔
Add any missing entity framework CRUD unit tests for Gift. ✔
Implement an Automapper Configuration between two Gift objects. ✔
Do not map the Id ✔
Create services for (do not map primary key Id for all entities):
Groups ✔
Users ✔
Gifts ✔
When retrieving the Gift object from the service, User should be pre-populated (incuded) ✔
Refactor out SampleData (see BlogEngine's SampleData.cs) ✔
Extra Credit
The following are options for extra credit (you don't need to do them all):

Write exceptional code that is clean, conforms to all coding guidelines, is well refactored, and easy to understand.

Well done!

File: SecretSanta/test/SecretSanta.Business.Tests/TestBase.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using AutoMapper;
```
Comment (MarkMichaelis) [inline SecretSanta/test/SecretSanta.Business.Tests/TestBase.cs]: I would love to have seen this class as a generic class containing more class refactored out of the other classes.

**Comment by MarkMichaelis** [comment]
# Assignment 3

## Main Assignment

- Connect Azure DevOps up for your fork.
- Add a Status Badge (see [Add a status badge to your repository](https://docs.microsoft.com/en-us/azure/devops/pipelines/create-first-pipeline?view=azure-devops&tabs=browser%2Ctfs-2018-2#add-a-status-badge-to-your-repository)) to the `README.md` file in `master` that shows the build status on each branch. ✔
- Be sure that clicking on the badge successfully navigates one to the Azure DevOps pipeline. ✔
- Rebase to the Readme.md update on `master` to the `Assignment3` branch so it too shows the badge (be sure it points to the Assignment3 branch status) ✔
- Appropriately encapsulate Id and all fingerprint data. ✔
- Add non-default constructors to Gift and address all warnings (see BlogEngine's Post.cs file). ✔
- Add any missing entity framework CRUD unit tests for `Gift`. ✔
- Implement an Automapper Configuration between two `Gift` objects. ✔
  - Do not map the Id ✔
- Create services for (do not map primary key Id for all entities):
  - Groups ✔
  - Users ✔
  - Gifts ✔
  - When retrieving the Gift object from the service, User should be pre-populated (included) ✔
- Refactor out SampleData (see BlogEngine's SampleData.cs) ✔

## Extra Credit

The following are options for extra credit (you don't need to do them all):

- Write exceptional code that is clean, conforms to all coding guidelines, is well refactored, and easy to understand. 

## Project

| ProjectName | Nullable | Solutions | Analyzers | Unexpected Disabled Warnings | Waning Level |
| -- | -- | -- | -- | -- | -- |
| SecretSanta.Api | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Business | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Data | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Web | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Api.Tests | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Business.Tests | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Data.Tests | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Web.Tests | Enable | SecretSanta.sln |  |  | 4 |


| Project Analysis |   |
| -----------------| - |
| Build/Compile | (Build succeeded: 0 Error(s), 1 Warning(s)) |
| Disabled Warnings **** | **✔**
| Nullable Enabled: | **✔** |
| Appropriately handle all warnings | **❌** |
| Test Run Succeeded: Total tests=18, Passed=18, Failed=0, Skipped=0 |

**Pull request closed without merging**